libsolidity/ast/Types.h

```diff
@@ -208,14 +208,26 @@ class Type
 	virtual bool operator==(Type const& _other) const { return category() == _other.category(); }
 	virtual bool operator!=(Type const& _other) const { return !this->operator ==(_other); }
 
-	/// @returns number of bytes used by this type when encoded for CALL. If it is a dynamic type,
-	/// returns the size of the pointer (usually 32). Returns 0 if the type cannot be encoded
+	/// @returns number of bytes used by this type when encoded for CALL. Cannot be used for
+	/// dynamically encoded types. Returns 0 if the type cannot be encoded
```

Comment:
Yes, that's a good idea - I'll look into whether we can just do it in this PR here as well - I think there is no case that doesn't throw anyways, which calls this when it would return zero - if that's true we can just make it an assertion right away.

Reply:
My last commit let's all three functions throw, if the type cannot be encoded in calldata.